packages/react-input/src/components/Input/Input.tsx

```diff
@@ -7,7 +7,7 @@ import type { InputProps } from './Input.types';
 /**
  * Input component
  */
-export const Input = React.forwardRef<HTMLElement, InputProps>((props, ref) => {
+export const Input: React.FunctionComponent<InputProps> = React.forwardRef((props, ref) => {
```

Comment:
I don't think that this typing is correct as `React.FunctionComponent` does not have `ref` on it's interface:

```
Type '{ foo: string; ref: RefObject<unknown>; }' is not assignable to type 'IntrinsicAttributes & Props & { children?: ReactNode; }'.
  Property 'ref' does not exist on type 'IntrinsicAttributes & Props & { children?: ReactNode; }'.
```

[TypeScript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQLABQ9MAnmFnAArFjoC8cA3nAIQIALjioYUYADsA5nAC+9erggzJcAJIywAVxgBGOH2x4YAOmFQA7sigATbAQA8ACQAqAWQAyAEWAANwBRABssECwZGAAaDi5UAD4ACmSwBLjKAgBKE0S4FwcggQtS9IhuRTgsnn4spTgAekTs1XVNHX0YACZxM3wLADE9GXxgdQBhEkgZKJgXTgqkk0wqS2s7R2d3b38gsIi5uMXuFLSM6qwcvIKiwJKyhKqauquG5taGOgIRsfU4DxYSTJXL8eiUGB6KAyODJehwAqJeEIgqdAzGYQQHhkTFkS4EWr9Sy4SjIGBYZwgqrNZEIlxonpCETY3H4wlrCwknDkynZalIuh0ml0T70RRAA)

Reply:
`InputProps` has `ref` through inheritance. I strongly prefer an explicit type for the component here because it makes the public API cleaner.